tests/test_utils.py

```diff
@@ -105,6 +105,35 @@ def test_class_instantiate_passthrough(
     assert obj == expected
 
 
+@pytest.mark.parametrize(  # type: ignore
+    "input_conf, expected",
+    [
+        (
+            {
+                "all_params": {
+                    "main": {
+                        "class": "tests.test_utils.Bar",
+                        "params": {"a": 10, "b": 20, "c": "${all_params.aux.c}"},
+                    },
+                    "aux": {"c": 30},
+                }
+            },
+            Bar(10, 20, 30, 40),
+        ),
+    ],
+)
+def test_string_iterpolation_during_deepcopy(
+    input_conf: Dict[str, Any], expected: Any
+) -> None:
+    # Check if the instantiate method maintains the parent when making a deepcopy
+    conf = OmegaConf.create(input_conf)
+    orig = copy.deepcopy(conf)
```

Comment:
I don't think you need to do the deepcopy here.
Instantiate is doing the deepcopy.

Reply:
You are also testing passthrough.
just make your test another case of test_class_instantiate.
(You can add additional parameters set to the list there).

Reply:
I can get rid of the deepcopy part. Regarding adding extra params to `test_class_instantiate`, I think it will make it a little messy as we need a nested config and need to call `instantiate` over a specific node in the config. If that is fine, I can merge the two testcases.

Reply:
As long as the test code is the same (and it can be), you can reduce duplication with parametrize.

Reply:
Merged the test cases